### ISSUE TITLE
Black screen on some games

### ISSUE THREAD

**Issue by lollo78** [libretro/libretro-vecx#3]
Hi!
I tried different roms, but games like 3D crazy coaster, 3D Minestorm and 3D Narrow Escape doesn't works. The games start (I can see the Vectrex intro), but after the screen remains black.
Any suggestion?
Thx a lot

**Comment by orbea** [comment]
@lollo78 What OS, gpu and driver were you using? Can you still reproduce this?

I can reproduce this with Slackware using nouveau (mesa). It seems the game requires `glXSwapBuffersMscOML` which my driver does not support, I imagine yours doesn't either.
```
3735 @0 glXSwapBuffersMscOML(dpy = 0x109b4f0, drawable = 52428803, target_msc = 0, divisor = 0, remainder = 0) = 1
3735: warning: unsupported glXSwapBuffersMscOML call
3815 @0 glXSwapBuffersMscOML(dpy = 0x109b4f0, drawable = 52428803, target_msc = 0, divisor = 0, remainder = 0) = 2
3815: warning: unsupported glXSwapBuffersMscOML call
3873 @0 glXSwapBuffersMscOML(dpy = 0x109b4f0, drawable = 52428803, target_msc = 0, divisor = 0, remainder = 0) = 3
3873: warning: unsupported glXSwapBuffersMscOML call
```
Apitrace - http://ks392457.kimsufi.com/orbea/stuff/trace/libretro/vecx_3D_glXSwapBuffersMscOML.trace.xz
Apitrace log - http://pastebin.com/30kQEaBH

**Comment by orbea** [comment]
Actually that is a RetroArch problem and not vecx...

**Comment by lollo78** [comment]
I'm waiting a new computer. Mine is exploded. :(
For what I can remember the issue affects the "3d" titles.
I tried it on rpi3, Ubuntu 16.10 and windows 10.

**Comment by 777sha** [comment]
same issues with the "3D" games

**Comment by hizzlekizzle** [comment]
It looks like 3D-Imager games require special handling, and there's no such support in the codebase (i.e., neither in the libretro-ization nor in standalone/upstream) AFAICT.

**Comment by klepp0906** [comment]
I assume this still hasnt been solved? as im having the same issue as we speak, so far in all games ive tried.

**Comment by hizzlekizzle** [comment]
have you tried Mine Storm (1 or 2)? Those should definitely work, as they're the ones I test most of the time.

**Comment by klepp0906** [comment]
> have you tried Mine Storm (1 or 2)? Those should definitely work, as they're the ones I test most of the time.

yes, i did finally get to those/ones that work.   At least I know the core itself is set up and functioning properly.  As for the rest - it looks like its gonna be no bueno until the "special" handling is addressed :)

**Comment by lmshaker** [comment]
Sorry if this is TMI, but it's probably the cause of this issue with the 3D Games:

The Vectrex 3D Imager is a set of goggles, that contain a spinning disk that you look through.
The disk is 1/2 blacked out, so as the disk spins, you can see out of only one eye at a time.
 
The Vectrex 3D Imager is plugged into the Player 2 Controller Port. 
The Player 2 Port powers the 3D Imager. 
*And* the 3D Imager simulates button presses on the Player 2, Button 4, so the Vectrex can time frames for each eye.

No Player 2, Button 4 presses = No image.
